config/mail.global.php.dist

```diff
@@ -26,88 +26,57 @@ return [
              *
              * defaults to sendmail
              **/
-
-            'transport' => Symfony\Component\Mailer\Transport\SendmailTransport::class,
-
+            'transport' => SendmailTransport::class,
```

Comment:
again, 2 different places to set the transport?

Reply:
> again, 2 different places to set the transport?

This is the only place where `transport` is set.
The code on line 5 is just a `use` so we can don't set the FQCN on line 29.
This is a requirement set by PHPCS.

Reply:
that mean at line 5 we need to add `both` possible transports

Reply:
Nope.
The other transport (SmtpTransport) is only mentioned in comments ([L25](https://github.com/dotkernel/dot-mail/blob/issues-70/config/mail.global.php.dist#L25) and [L63](https://github.com/dotkernel/dot-mail/blob/issues-70/config/mail.global.php.dist#L63)).
Comments are ignored by PHPCS.

If we were to change the default transport, then we would replace:
* on line `5`: `use Symfony\Component\Mailer\Transport\SendmailTransport;` with `use Symfony\Component\Mailer\Transport\Smtp\SmtpTransport;`
* on line `29`: `SendmailTransport::class` with `SmtpTransport::class`

Reply:
I want to change the transport only in one place.
Why not have on line 5 both transports available ?
use sendmail,
use smtp

And decide on line 29 only which one to actually use

Reply:
As long as we do not use both classes in this file, we cannot add a `use` for both of them, one on line `5` and another on line `6` because PHPCS will fail.

Running:
```shell
composer cs-check
```
will output:
```text
> phpcs
.E........................................... 45 / 45 (100%)



FILE: config/mail.global.php
------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------
 6 | ERROR | [x] Type Symfony\Component\Mailer\Transport\Smtp\SmtpTransport is not used in this file.
------------------------------------------------------------------------------------------------------
```

Switching transports using an IDE (vs text editor) will autocomplete with the correct `use`, then the developer needs to remove the other - un`use`d - class, else both the IDE and PHPCS will complain that `Type x is not used in this file`.

Reply:
at least at line 5 we shod have both classes, one commented and one not commented
now is completely confusing what transport is used and how to switch to another one

Reply:
It should not be confusing because you should not look at the `use`s at the beginning of the file but at the value of the  `transport` key.
Again, in an IDE, you will not have this confusion because it will automatically inform you about missing/extra use statements.
Also, being a `.global` file, there shouldn't be any case where you need to manually edit (using a text editor, that does not provide checks) that file on a staging/production server. If you must though, you can create a `.local` copy of it.